packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -2369,3 +2409,40 @@ void main() {
     expect(find.text(snackBars[2]), findsNothing);
   });
 }
+
+/// Create drag gestures for DismissDirections.
+Map<DismissDirection, List<Offset>> _getDragGesturesOfDismissDirections(double scaffoldWidth) {
+  final Map<DismissDirection, List<Offset>> dragGestures = <DismissDirection, List<Offset>>{};
+
+  for (final DismissDirection val in DismissDirection.values) {
+    switch (val) {
+      case DismissDirection.down:
+        dragGestures[val] = <Offset>[const Offset(0.0, 50.0)]; // drag to bottom gesture
+        break;
+      case DismissDirection.up:
+        dragGestures[val] = <Offset>[const Offset(0.0, -50.0)]; // drag to top gesture
+        break;
+      case DismissDirection.vertical:
+        dragGestures[val] = <Offset>[
+          const Offset(0.0, 50.0), // drag to bottom gesture
+          const Offset(0.0, -50.0), // drag to top gesture
+        ];
+        break;
+      case DismissDirection.startToEnd:
+        dragGestures[val] = <Offset>[Offset(scaffoldWidth, 0.0)]; // drag to right gesture
+        break;
+      case DismissDirection.endToStart:
```

Comment:
Does this account for `TextDirection`? It may be worth adding a test that used `TextDirection.ltr` versus `TextDirection.rtl` to ensure the behavior is correct.

Reply:
If it's to test the `Dismissible` widget, I think it is already considered in `dismissible_test.dart`.